pim/views/product_view.xml

```diff
@@ -26,17 +26,24 @@
                 />
                 <field name="attribute_set_id" />
                 <field name="attribute_set_not_completed_ids" invisible="1" />
+                <field name="attribute_set_completeneness_ids" invisible="1" />
                 <field
                     widget='label_selection'
                     name="completion_state"
                     options="{'classes': {'complete': 'success', 'not_complete':'danger'}}"
-                    attrs="{'invisible': ['|', ('attribute_set_id', '=', False), ('attribute_set_not_completed_ids', '=', [])]}"
+                    attrs="{'invisible': [
+                                            '|',
+                                            ('attribute_set_id', '=', False),
+                                            '&amp;',
+                                            ('attribute_set_not_completed_ids', '=', []),
+                                            ('attribute_set_completeneness_ids', '=', []),
```

Comment:
To avoid complex domain, you could also add a computed field `is_completion_state_visible` that implement the logic on the python side. In this way, you'll avoid to duplicate the same logic into the product.product and product.template form.